modules/database/src/adapters/sequelize-adapter/utils/schema.ts

```diff
@@ -136,12 +137,29 @@ export async function getTransactionAndParsedQuery(
   } else {
     parsedQuery = query;
   }
-  if (parsedQuery.hasOwnProperty('$set')) {
-    parsedQuery = parsedQuery['$set'];
+  if (method === 'PATCH') {
+    if (!parsedQuery.hasOwnProperty('$set')) {
+      parsedQuery = { $set: parsedQuery };
+    } else {
+      parsedQuery = parsedQuery['$set'];
+    }
+  }
+  if (method === 'PUT') {
+    if (parsedQuery.hasOwnProperty('$set')) {
+      parsedQuery = parsedQuery['$set'];
+    } else {
+      const fields = Object.keys(parsedQuery);
+      for (const field of fields) {
```

Comment:
It's pointless to iterate on the fields passed. Your for loop should iterate on the fields that the schema we're trying to modify contains, and check if a field does not exist in the query, then set it to null